### ISSUE TITLE
error in making a copy of a model

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#237]
I'm seeing this at the end of Chapter 5 of the User Manual under 0.6-1. Don't have time to investigate at the moment.

```
simpleModelCopy3 <- simpleModelCopy2$newModel(check = FALSE)
## setting data and initial values...
## Error in dataNames[i] <- data[[i]]:
in subassignment type fix
```

**Comment by paciorek** [comment]
Just verified that this worked as of 0.5-1 and did not work starting with 0.6.

**Comment by perrydv** [comment]
This is a trivial bug.  What happens is that model$newModel attempts to use the origData that was provided when the first model was build.  If nothing was provided that time, this is a uninitialized field, which causes a bug for building the new model.  In the future we can catch that case.  For immediate purposes of building a valid manual, we can use:

simpleModelCopy3 <- simpleModelCopy2$newModel(check = FALSE, data = list())

Would probably be a good idea to add tests for newModel and possibly other utilities that we don’t use much ourselves.

**Comment by perrydv** [comment]
We fixed this for 0.6-4.
